kore/src/Kore/Internal/Predicate.hs

```diff
@@ -734,6 +738,12 @@ setSimplified
 setSimplified simplified (GenericPredicate termLike) =
     GenericPredicate (TermLike.setSimplified simplified termLike)
 
+forgetSimplified
+    :: InternalVariable variable
+    => Predicate variable -> Predicate variable
+forgetSimplified (GenericPredicate termLike) =
+    GenericPredicate (resynthesize termLike)
```

Comment:
Why not have a `TermLike.forgetSimplified` which calls `resynthesize`?

Right now `resynthesize` does the right thing because the `Simplified` attribute is the only one that cannot be recomputed from the term, but that may change in the future, and it would be nice to have a single place to change when that happens.

Reply:
Done!